### PR TITLE
MULE-18894: NotSerializableException when persisting ABP decorators.

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java [hunk 1]
```diff
@@ -44,4 +58,15 @@ public Spliterator spliterator() {
     populator.accept(size());
     return super.toArray(object);
   }
+
+  @Override
+  public void writeExternal(ObjectOutput out) throws IOException {
+    out.writeObject(getCollection());
+
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java:65]: remove empty line

File: core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java [hunk 2]
```diff
@@ -44,4 +58,15 @@ public Spliterator spliterator() {
...
+  }
+
+  @Override
+  public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+    this.collection = (Collection) in.readObject();
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java:70]: we shouldn't be expecting to encounter this actual class serialized.

File: core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsList.java [hunk 3]
```diff
@@ -45,4 +61,14 @@ public Spliterator spliterator() {
...
   }
 
+  @Override
+  public void writeExternal(ObjectOutput out) throws IOException {
+    out.writeObject(getList());
+  }
+
+  @Override
+  public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+    this.collection = (List) in.readObject();
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsList.java:71]: we shouldn't be expecting to encounter this actual class serialized.

File: core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java [hunk 4]
```diff
@@ -6,18 +6,32 @@
...
 import java.util.function.LongConsumer;
 
 import org.apache.commons.collections.collection.AbstractCollectionDecorator;
 
-class PayloadStatisticsCollection<T> extends AbstractCollectionDecorator {
+class PayloadStatisticsCollection<T> extends AbstractCollectionDecorator implements Externalizable {
 
   private final LongConsumer populator;
 
+  public PayloadStatisticsCollection() {
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java:26]: would this be needed if there were a kryo serializer for this class?

File: core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsList.java [hunk 5]
```diff
@@ -14,10 +19,21 @@
...
 import org.apache.commons.collections.list.AbstractListDecorator;
 
-final class PayloadStatisticsList<T> extends AbstractListDecorator {
+final class PayloadStatisticsList<T> extends AbstractListDecorator implements Externalizable {
+
+  private static final long serialVersionUID = -5991764233175594789L;
 
   private final LongConsumer populator;
 
+  public PayloadStatisticsList() {
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsList.java:28]: would this be needed if there were a kryo serializer for this class?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java (hunk 4 above)
Comment (fsgonz) [reply core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java:26]: the void constructor is needed for non-kryo scenarios. Kryo also needs all parameters to be initialized in the void constructor.

File: core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsList.java (hunk 5 above)
Comment (fsgonz) [reply core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsList.java:28]: the void constructor is needed for non-kryo scenarios. Kryo also needs all parameters to be initialized in the void constructor.

File: core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java (hunk 2 above)
Comment (fsgonz) [reply core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java:70]: Needed for the case where we are not using kryo.

**Review by elrodro83 — APPROVED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**